dm/master/server_test.go

```diff
@@ -1288,7 +1295,9 @@ func (t *testMaster) TestSwitchWorkerRelayMaster(c *check.C) {
 	c.Assert(resp.Workers, check.HasLen, 2)
 	for _, w := range resp.Workers {
 		c.Assert(w.Result, check.IsFalse)
-		c.Assert(w.Msg, check.Matches, ".*relevant worker-client not found")
+		lines := strings.Split(w.Msg, "\n")
+		c.Assert(len(lines), check.Greater, 1)
```

Comment:
why do we check this?

Reply:
w.Msg has error stack here, check the cause error message only

Reply:
how about using `.*relevant worker-client not found.*`?

Reply:
'\n' is not matched

Reply:
I will refine it, use multi-line pattern regex expression